daemon/src/to_sse_event.rs

```diff
@@ -146,20 +126,11 @@ impl CfdsWithAuxData {
     }
 
     /// Check whether given CFD has any active settlement proposals
-    fn settlement_proposal_status(&self, cfd: &model::cfd::Cfd) -> SettlementProposalStatus {
-        match &self.settlement_proposals {
-            SettlementProposals::Incoming(proposals) => {
-                if proposals.contains_key(&cfd.order.id) {
-                    return SettlementProposalStatus::Incoming;
-                }
-            }
-            SettlementProposals::Outgoing(proposals) => {
-                if proposals.contains_key(&cfd.order.id) {
-                    return SettlementProposalStatus::Outgoing;
-                }
-            }
+    fn settlement_proposal_status(&self, cfd: &model::cfd::Cfd) -> Option<SettlementKind> {
+        if let Some((_, kind)) = self.settlement_proposals.get(&cfd.order.id) {
+            return Some(kind.clone());
         }
-        SettlementProposalStatus::None
+        None
```

Comment:
This could be replaced by a `.map` on the `Option`.

Reply:
thanks!  I still sometimes forget about these cool features ;)

changed to:
```rust
       self.settlement_proposals
           .get(&cfd.order.id)
           .map(|(_, kind)| kind.clone())
```